src/features/search/helpers/categoriesHelpers/categoriesHelpers.ts

```diff
@@ -269,3 +270,32 @@ export function getDataByAlphabeticalSorting(
 ) {
   return data.sort((a, b) => (a?.value || '').localeCompare(b?.value || ''))
 }
+
+export function isOnlyOnline(
+  data: SubcategoriesResponseModelv2,
+  categoryId?: SearchGroupNameEnumv2,
+  nativeCategoryId?: NativeCategoryIdEnumv2
+) {
+  if (!categoryId && !nativeCategoryId) {
+    return undefined
+  }
+
+  const platforms: OnlineOfflinePlatformChoicesEnumv2[] = [
+    ...new Set(
+      data.subcategories
+        .filter((subcategory) =>
+          nativeCategoryId
+            ? subcategory.nativeCategoryId === nativeCategoryId
+            : subcategory.searchGroupName === categoryId
+        )
+        .map((subcategory) => subcategory.onlineOfflinePlatform)
+    ),
+  ]
+
+  const isOnlyOnline =
+    platforms.includes(OnlineOfflinePlatformChoicesEnumv2.ONLINE) &&
+    !platforms.includes(OnlineOfflinePlatformChoicesEnumv2.ONLINE_OR_OFFLINE) &&
+    !platforms.includes(OnlineOfflinePlatformChoicesEnumv2.OFFLINE)
+
+  return isOnlyOnline ? isOnlyOnline : undefined
```

Comment:
Pourquoi pas juste ?
```suggestion
  return isOnlyOnline
```

Tu retournerais un `boolean` plutôt que `true | undefined`

Reply:
Yes mais je me suis dis que ça surcharge l'URL pour rien quand il est `false`

Reply:
Ah oui je vois. L'idée est bonne alors mais je te conseille de faire ça ailleurs : par exemple à l'endroit où tu le mets dans l'URL.
Je m'attends de prime abord à ce que `isOnlyOnline` retourne un boolean :D 

Enfin c'est un détail. Je peux accepter quand même

Reply:
Yes je comprends ton point de vue, tu préfères quelque chose comme ça ?

  ```
      let isOnline = false
      if (payload.offerNativeCategories.length > 0) {
        isOnline = isOnlyOnline(data, undefined, payload.offerNativeCategories[0])
      } else if (payload.offerCategories.length > 0) {
        isOnline = isOnlyOnline(data, undefined, payload.offerNativeCategories[0])
      } else {
        isOnline = false
      }
      additionalSearchState = {
        ...additionalSearchState,
        ...(isOnline ? { isOnline } : {}),
      }
```

Reply:
Ca marcherait pas ça plutôt ?
```ts
additionalSearchState = {
  ...additionalSearchState,
  isOnline: isOnline || undefined,
}
```
Ce serait plus facile à lire je pense :D
Mais dans l'idée, je préfère mieux oui !
Tu fais comme tu le sens du coup :D